src/main/java/ua/lviv/iot/models/Performance.java

```diff
@@ -0,0 +1,24 @@
+package ua.lviv.iot.models;
+
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+
+@Getter
+@Setter
+@NonNull
+@AllArgsConstructor
+@FieldDefaults(makeFinal = false, level = AccessLevel.PROTECTED)

```

Comment:
Interesting, why some part of your classes use AccessLevel.PRIVATE and this one is PROTECTED?

Reply:
This class uses PROTECTED for inheritance realization, because it is father class and as soon as other classes aren't supposed to have children, they use PRIVATE and are final

Reply:
Also, access modifiers are the same as in UML diagram